src/components/BottomMenu.jsx

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from "react";
+import { SafeAreaView, StyleSheet, TouchableOpacity, View } from "react-native";
+import HomeIcon from "./Home";
+import MentalIcon from "./Mental";
+import GrowIcon from "./Grow";
+import SettingsIcon from "./Settings";
+
+const BottomMenu = () => {
+  const [selectedButton, setSelectedButton] = useState(null);
```

Comment:
it is called selectedButton, but actually you are using indexes, this might be problematic because order may change, and buttons may change.

so instead, use javascript enums to define all possible values (including null value, which could be named notChosen or something similiar), please ask me if you dont understand

https://www.geeksforgeeks.org/enums-in-javascript/

Reply:
and for names, please be consistent with icon names, so home icon would be home button, and states could be home, mental, growth.

please try to name these states (enums) the best you can, as they also define names for the views, they should all be consistent